File: Check Palindrome Number/check_palindrome_number.py
n = int(input("Enter a number: "))
original = n
reverse = 0
while n > 0:
    digit = n % 10
    reverse = reverse * 10 + digit
    n //= 10
if reverse == original:
    print(original, "is a palindrome number")
else:
    print(original, "is not a palindrome number")

File: Check Palindrome Number/CheckPalindromeNumber.java
public class CheckPalindromeNumber {
    public static void main(String[] args) {
        java.util.Scanner sc = new java.util.Scanner(System.in);
        System.out.print("Enter a number: ");
        int n = sc.nextInt();
        int original = n, reverse = 0;
        while (n > 0) {
            int digit = n % 10;
            reverse = reverse * 10 + digit;
            n = n / 10;
        }
        if (reverse == original) {
            System.out.println(original + " is a palindrome number");
        } else {
            System.out.println(original + " is not a palindrome number");
        }
    }
}
